### PR TITLE
MOTOR-1054 motor.motor_asyncio is broken for Python 3.11

### PR DESCRIPTION
**Pull request by blink1073**

### CODE REVIEW

File: doc/changelog.rst [hunk 1]
```diff
@@ -3,6 +3,13 @@ Changelog
 
 .. currentmodule:: motor.motor_tornado
 
+Motor 3.1.1
+-----------
+
+Motor 3.3.1 adds support for Python 3.11 and fixes a bug that caused an
+``ImportError`` that version.
```
Comment (behackett) [inline doc/changelog.rst:10]: with that version

File: doc/changelog.rst [hunk 2]
```diff
@@ -3,6 +3,13 @@ Changelog
 
 .. currentmodule:: motor.motor_tornado
 
+Motor 3.1.1
+-----------
+
+Motor 3.3.1 adds support for Python 3.11 and fixes a bug that caused an
```
Comment (behackett) [inline doc/changelog.rst:9]: 3.3.1 or 3.1.1?

File: .evergreen/config.yml [hunk 3]
```diff
@@ -843,10 +847,14 @@ axes:
...
            TOX_ENV: "asyncio-py39"
            PYTHON_BINARY: "/opt/python/3.9/bin/python3"
+      - id: "asyncio-py310"
+        variables:
+           TOX_ENV: "asyncio-py310"
+           PYTHON_BINARY: "/opt/python/3.10/bin/python3"
       - id: "py3-pymongo-latest"
         variables:
            TOX_ENV: "py3-pymongo-latest"
-           PYTHON_BINARY: "/opt/python/3.7/bin/python3"
```
Comment (behackett) [inline .evergreen/config.yml:849]: Is this supposed to be the min version?

**Review by behackett — CHANGES_REQUESTED** [review]

File: .evergreen/config.yml (hunk 3 above)
Comment (ShaneHarvey) [reply .evergreen/config.yml:849]: Agreed. I think we should keep this as the min version we support for now. This test fails on 3.10:
```
 [2022/10/25 17:32:56.873] ImportError: Failed to import test module: tornado_tests
 [2022/10/25 17:32:56.873] Traceback (most recent call last):
 [2022/10/25 17:32:56.873]   File "/opt/python/3.10/lib/python3.10/unittest/loader.py", line 470, in _find_test_path
 [2022/10/25 17:32:56.873]     package = self._get_module_from_name(name)
 [2022/10/25 17:32:56.873]   File "/data/mci/593377d0ad434cdc1766d60588588194/src/test/__init__.py", line 72, in _get_module_from_name
 [2022/10/25 17:32:56.873]     return super()._get_module_from_name(name)
 [2022/10/25 17:32:56.873]   File "/opt/python/3.10/lib/python3.10/unittest/loader.py", line 377, in _get_module_from_name
 [2022/10/25 17:32:56.873]     __import__(name)
 [2022/10/25 17:32:56.873]   File "/data/mci/593377d0ad434cdc1766d60588588194/src/test/tornado_tests/__init__.py", line 25, in <module>
 [2022/10/25 17:32:56.873]     from tornado import testing
 [2022/10/25 17:32:56.873]   File "/data/mci/593377d0ad434cdc1766d60588588194/src/.tox/py3-pymongo-latest/lib/python3.10/site-packages/tornado/testing.py", line 16, in <module>
 [2022/10/25 17:32:56.873]     from tornado.httpclient import AsyncHTTPClient
 [2022/10/25 17:32:56.873]   File "/data/mci/593377d0ad434cdc1766d60588588194/src/.tox/py3-pymongo-latest/lib/python3.10/site-packages/tornado/httpclient.py", line 50, in <module>
 [2022/10/25 17:32:56.873]     from tornado import gen, httputil, stack_context
 [2022/10/25 17:32:56.873]   File "/data/mci/593377d0ad434cdc1766d60588588194/src/.tox/py3-pymongo-latest/lib/python3.10/site-packages/tornado/httputil.py", line 107, in <module>
 [2022/10/25 17:32:56.873]     class HTTPHeaders(collections.MutableMapping):
 [2022/10/25 17:32:56.873] AttributeError: module 'collections' has no attribute 'MutableMapping'
```

File: doc/requirements.rst [hunk 4]
```diff
@@ -135,32 +141,36 @@ Motor 2.5 deprecated support for Python 3.5.
 
 Motor 3.0 dropped support for Pythons older than 3.7.
 
+Motor 3.1.1 added support for Python 3.11.
+
 +-------------------------------------------------------------------------------------+
 |                   Python Version                                                    |
 +=====================+=====+=====+=====+=======+=======+=====+=====+=====+=====+=====+
```
Comment (ShaneHarvey) [inline doc/requirements.rst:148]: the `+======` and `+------` lines need to be extended for the new 3.11 column: https://github.com/mongodb/motor/actions/runs/3322825314/jobs/5492384749

File: doc/changelog.rst (hunk 1 above)
Comment (blink1073) [reply doc/changelog.rst:10]: Fixed

File: .evergreen/config.yml (hunk 3 above)
Comment (blink1073) [reply .evergreen/config.yml:849]: Fixed

**Review by ShaneHarvey — APPROVED** [review]

**Review by behackett — APPROVED** [review]

**Comment by JWCook** [comment]
The fact that motor 3.1.1 is a python 3.11 compatibility patch is incredibly satisfying.

**Comment by behackett** [comment]
LOL. I didn't catch that before. Nice.

**Pull request merged**